dotCMS/src/main/java/com/dotmarketing/business/web/UserWebAPIImpl.java

```diff
@@ -16,55 +13,65 @@
  */
 public class UserWebAPIImpl extends UserAPIImpl implements UserWebAPI {
 
-	public UserWebAPIImpl() {
-		
-	}
+    public UserWebAPIImpl() {
+
+    }
+
 
-	@CloseDBIfOpened
+
+    @CloseDBIfOpened
     @Override
     public User getUser(HttpServletRequest request) {
-      try{
-        User user = this.getLoggedInUser(request);
-        if(user==null){
-          user =  APILocator.getUserAPI().getAnonymousUser();
+
+        User user = PortalUtil.getUser(request);
+
+        if (user == null) {
+            user = APILocator.getUserAPI().getAnonymousUserNoThrow();
         }
         return user;
-      }catch(Exception e){
-        throw new DotStateException(e);
-      }
+
+    }
+
+
+    @CloseDBIfOpened
+    @Override
+    public User getLoggedInUser(HttpServletRequest request) {
+        User user = PortalUtil.getUser(request);
+
+        return (user != null && !user.isAnonymousUser() && user.isActive() && (user.isBackendUser() || user.isFrontendUser()))
+                        ? user
+                        : null;
+    }
+
+    @Deprecated
+    @Override
+    public User getLoggedInUser(final HttpSession session) {
+        User user = PortalUtil.getUser(session);
+        return (user != null && !user.isAnonymousUser() && user.isActive() && (user.isBackendUser() || user.isFrontendUser()))
+                        ? user
+                        : null;
+    }
+    
+    @CloseDBIfOpened
+    @Override
+    public boolean isLoggedToBackend(HttpServletRequest request) {
+        return PortalUtil.getUser(request) != null && PortalUtil.getUser(request).isBackendUser();
     }
     
-	@Override
-	public User getLoggedInUser(HttpServletRequest request) {
-		User user = PortalUtil.getUser(request);
-		return (user == null)?
-				//Assuming is a front-end access
-				this.getLoggedInUser(request.getSession(false)):user;
-	}
-
-	@Override
-	public User getLoggedInUser(final HttpSession session) {
-
-		return  PortalUtil.getUser(session);
-	}
-
-	@Override
-	public boolean isLoggedToBackend(HttpServletRequest request) {
-		return PortalUtil.getUser(request) != null;
-	}
-
-	@Override
-	public User getLoggedInFrontendUser(HttpServletRequest request)  {
-		HttpSession session = request.getSession(false);
-		if(session != null)
-			return (User) session.getAttribute(WebKeys.CMS_USER);
-		return null;
-	}
-
-	@Override
-	public boolean isLoggedToFrontend(HttpServletRequest req)  {
-		return !isLoggedToBackend(req);
-	}
+    @CloseDBIfOpened
+    @Override
+    public User getLoggedInFrontendUser(HttpServletRequest request) {
+        User user = getLoggedInUser(request);
+
+        return user != null && PortalUtil.getUser(request).isFrontendUser() ? user : null;
+    }
+    
+    @CloseDBIfOpened
+    @Override
+    public boolean isLoggedToFrontend(HttpServletRequest request) {
+        User user = getLoggedInUser(request);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'user' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5106108)